ui/job-view/pushes/Push.jsx

```diff
@@ -38,11 +41,55 @@ export default class Push extends React.Component {
   componentDidMount() {
     this.showRunnableJobs = this.showRunnableJobs.bind(this);
     this.hideRunnableJobs = this.hideRunnableJobs.bind(this);
+    this.toggleSelectedRunnableJob = this.toggleSelectedRunnableJob.bind(this);
     this.expandAllPushGroups = this.expandAllPushGroups.bind(this);
+    this.fetchJobs = this.fetchJobs.bind(this);
+    this.groupJobByPlatform = this.groupJobByPlatform.bind(this);
+    this.sortGroupedJobs = this.sortGroupedJobs.bind(this);
+    this.mapPushJobs = this.mapPushJobs.bind(this);
+    this.poll = this.poll.bind(this);
+    this.getLastModifiedJobTime = this.getLastModifiedJobTime.bind(this);
+
+    this.fetchJobs();
+    this.poll();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    this.showUpdateNotifications(prevState);
+  }
+
+  componentWillUnmount() {
+    if (this.pollIntervalId) {
+      clearTimeout(this.pollIntervalId);
+      this.pollIntervalId = null;
+    }
+  }
+
+  getJobCount(jobList) {
+    return jobList.reduce((memo, job) => (
+      job.result !== 'superseded' ? { ...memo, [job.state]: memo[job.state] + 1 } : memo
+      ), { running: 0, pending: 0, completed: 0 },
+    );
+  }
+
+  getJobGroupInfo(job) {
+    const {
+      job_group_name: name, job_group_symbol,
+      platform, platform_option, tier,
+    } = job;
+    const symbol = job_group_symbol === '?' ? '' : job_group_symbol;
+    const mapKey = getGroupMapKey(
+      symbol, tier, platform, platform_option);
+
+    return { name, tier, symbol, mapKey };
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    this.showUpdateNotifications(nextProps);
+  getLastModifiedJobTime() {
+    const { jobList } = this.state;
+    const latest = max(jobList.map(job => new Date(job.last_modified + 'Z'))) || new Date();
+
+    latest.setSeconds(latest.getSeconds() - 3);
+    return latest;
   }
 
   componentDidCatch(error) {
```

Comment:
Did we decide to favor use of the ErrorBoundaries component instead of using this life cycle method in individual components?

Reply:
Another good catch.  Yeah, this logic is unnecessary now that pushes are wrapped in an ``ErrorBoundary``.  Removed.